FluentTc.Tests/AcceptanceTests.cs

```diff
@@ -1599,5 +1599,99 @@ public void SetBuildName(string name)
+
+            // Act
+            var vcsRoot = connectedTc.CreateVcsRoot(project, __ => __
+                .AgentCleanFilePolicy(AgentCleanFilePolicy.AllIgnoredUntrackedFiles)
+                .AgentCleanPolicy(AgentCleanPolicy.Always)
+                .AuthMethod(AuthMethod.Anonymous)
+                .Branch("refs/head/develop")
+                .BranchSpec("+:refs/head/feature/*")
+                .Id("VcsRootId")
+                .Name("VcsRootName")
+                .Password("Password")
+                .SubModuleCheckout(true)
+                .Url(new Uri("http://www.gooogle.com"))
+                .UseAlternates(true)
+                .Username("Username")
+                .UserNameStyle(UserNameStyle.AuthorName));
+
+            // Assert
+            string xmlData = string.Format(
+                @"<vcs-root id=""{0}"" name=""{1}"" vcsName=""{2}"">
+                    <project id=""{3}"" name=""{4}"" href=""{5}""/>
+                    <properties count =""{6}"">",
+                vcsRoot.Id, vcsRoot.Name, vcsRoot.vcsName, vcsRoot.Project.Id, vcsRoot.Project.Name, vcsRoot.Project.Href, vcsRoot.Properties.Property.Count);
+
+            foreach (var property in vcsRoot.Properties.Property)
+            {
+                xmlData += @"<property name=""";
+                xmlData += property.Name + @"""";
+                xmlData += @" value=""" + property.Value + @"""/>";
+            }
+            xmlData += @"</properties>";
+
+            xmlData += @"</vcs-root>";
+
+            A.CallTo(
+                () =>
+                    teamCityCaller.Post(xmlData,
+                    HttpContentTypes.ApplicationXml, "/app/rest/vcs-roots", HttpContentTypes.ApplicationJson))
+                        .MustHaveHappened(Repeated.Exactly.Once);
+        }
+
+        [Test]
+        public void AttachVcsRoot()
+        {
+            // Arrange
+            var teamCityCaller = CreateTeamCityCaller();
+
+            var connectedTc = new RemoteTc().Connect(_ => _.AsGuest(), teamCityCaller);
+
+            var project = connectedTc.GetProjectById("ProjectId");
+            var vcsRoot = connectedTc.CreateVcsRoot(project, __ => __
+                .AgentCleanFilePolicy(AgentCleanFilePolicy.AllIgnoredUntrackedFiles)
+                .AgentCleanPolicy(AgentCleanPolicy.Always)
+                .AuthMethod(AuthMethod.Anonymous)
+                .Branch("refs/head/develop")
+                .BranchSpec("+:refs/head/feature/*")
+                .Id("VcsRootId")
+                .Name("VcsRootName")
+                .Password("Password")
+                .SubModuleCheckout(true)
+                .Url(new Uri("http://www.gooogle.com"))
+                .UseAlternates(true)
+                .Username("Username")
+                .UserNameStyle(UserNameStyle.AuthorName));
+
+            // Act
+            connectedTc.AttachVcsRootToBuildConfiguration(_ => _.Id("BuildId"), vcsRoot);
```

Comment:
I meant to locate VCS root by Id. Checkout rules can be specified when VCS root is attached to build configuration. You don't have to implement it. It can be a separate issue. Whatever you decide